src/components/Layout.js

```diff
@@ -1,136 +1,129 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link } from 'gatsby';
 import Toggle from './Toggle';
 
 import { rhythm, scale } from '../utils/typography';
 import sun from '../assets/sun.png';
 import moon from '../assets/moon.png';
 
-class Layout extends React.Component {
-  state = {
-    theme: null,
-  };
-  componentDidMount() {
-    this.setState({ theme: window.__theme });
+function Layout(props) {
+  const [theme, setTheme] = useState(null);
+
+  useEffect(() => {
```

Comment:
I think this should be a layout effect.

Reply:
@gaearon I've just updated this in my latest commit